### PR TITLE
Calls to casts

### PR DESCRIPTION
**Pull request by jstypka**

### CODE REVIEW

File: src/concurrent/mas_conc_supervisor.erl [hunk 1]
```diff
@@ -46,36 +46,48 @@ close(Pid) ->
...
 
--spec handle_call(term(),{pid(),term()},state()) -> {reply,term(),state()} |
-                                                    {reply,term(),state(),hibernate | infinity | non_neg_integer()} |
-                                                    {noreply,state()} |
-                                                    {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                                    {stop,term(),term(),state()} |
-                                                    {stop,term(),state()}.
+-spec handle_call(term(),{pid(),term()},state()) ->
+                         {reply,term(),state()} |
+                         {reply,term(),state(),hibernate | infinity | non_neg_integer()} |
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Line 65 is too long:                          {reply,term(),state(),hibernate | infinity | non_neg_integer()} |.

File: src/concurrent/mas_conc_supervisor.erl [hunk 2]
```diff
@@ -46,36 +46,48 @@ close(Pid) ->
...
+                         {noreply,state()} |
+                         {noreply,state(),hibernate | infinity | non_neg_integer()} |
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Line 67 is too long:                          {noreply,state(),hibernate | infinity | non_neg_integer()} |.

File: src/concurrent/mas_conc_supervisor.erl [hunk 3]
```diff
@@ -46,36 +46,48 @@ close(Pid) ->
...
 handle_call(close, _From, St) ->
     [mas_conc_arena:close(Pid) || {_Name,Pid} <- dict:to_list(St#state.arenas)],
     {stop, normal, ok, St}.
 
--spec handle_cast(term(),state()) -> {noreply,state()} |
-                                     {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                     {stop,term(),state()}.
+-spec handle_cast(term(),state()) ->
+                         {noreply,state()} |
+                         {noreply,state(),hibernate | infinity | non_neg_integer()} |
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Line 76 is too long:                          {noreply,state(),hibernate | infinity | non_neg_integer()} |.

File: src/concurrent/mas_conc_supervisor.erl [hunk 4]
```diff
@@ -46,36 +46,48 @@ close(Pid) ->
...
+                             [A, St#state.arenas, SP, Cf]) || A <- Agents],
     {noreply, St}.
 
 
--spec handle_info(term(),state()) -> {noreply,state()} |
-                                     {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                     {stop,term(),state()}.
+-spec handle_info(term(),state()) ->
+                         {noreply,state()} |
+                         {noreply,state(),hibernate | infinity | non_neg_integer()} |
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Line 89 is too long:                          {noreply,state(),hibernate | infinity | non_neg_integer()} |.

**Comment by mpmiszczyk** [comment]
Code changes itself are good to be merged. Still I would like to talk about setting timeouts in gen_server; is it necessary, and how else it could be handled (supervision tree?).

File: src/concurrent/mas_conc_supervisor.erl [hunk 5]
```diff
@@ -46,36 +46,52 @@ close(Pid) ->
...
     {ok, #state{arenas = Arenas, config = Cf, sim_params = SP}}.
 
 
--spec handle_call(term(),{pid(),term()},state()) -> {reply,term(),state()} |
-                                                    {reply,term(),state(),hibernate | infinity | non_neg_integer()} |
-                                                    {noreply,state()} |
-                                                    {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                                    {stop,term(),term(),state()} |
-                                                    {stop,term(),state()}.
+-spec handle_call(term(),{pid(),term()},state()) ->
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Missing space after "," on line 63

File: src/concurrent/mas_conc_supervisor.erl [hunk 6]
```diff
@@ -46,36 +46,52 @@ close(Pid) ->
...
+                         {reply,term(),state()} |
+                         {reply,term(),state(),hibernate | infinity
+                          | non_neg_integer()} |
+                         {noreply,state()} |
+                         {noreply,state(),hibernate | infinity
+                          | non_neg_integer()} |
+                         {stop,term(),term(),state()} |
+                         {stop,term(),state()}.
 handle_call(close, _From, St) ->
     [mas_conc_arena:close(Pid) || {_Name,Pid} <- dict:to_list(St#state.arenas)],
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Missing space after "," on line 73

File: src/concurrent/mas_conc_supervisor.erl [hunk 7]
```diff
@@ -46,36 +46,52 @@ close(Pid) ->
...
     {stop, normal, ok, St}.
 
--spec handle_cast(term(),state()) -> {noreply,state()} |
-                                     {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                     {stop,term(),state()}.
+-spec handle_cast(term(),state()) ->
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Missing space after "," on line 76

File: src/concurrent/mas_conc_supervisor.erl [hunk 8]
```diff
@@ -46,36 +46,52 @@ close(Pid) ->
...
+    _InitPopulation = [spawn(mas_conc_agent,
+                             start,
+                             [A, St#state.arenas, SP, Cf]) || A <- Agents],
     {noreply, St}.
 
 
--spec handle_info(term(),state()) -> {noreply,state()} |
-                                     {noreply,state(),hibernate | infinity | non_neg_integer()} |
-                                     {stop,term(),state()}.
+-spec handle_info(term(),state()) ->
```
Comment (elvisci) [inline src/concurrent/mas_conc_supervisor.erl]: Missing space after "," on line 90

**Pull request merged**